src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -3280,7 +3280,10 @@ public void Medley()
                 "$(e.Substring($([System.IO.Path]::Combine(`a`, `b`))))",
                 "$([]::())",
                 "$((((",
+#if FEATURE_WIN32_REGISTRY
+                // If no registry, this gets expanded to the empty string
                 "$(Registry:X)",
+#endif
```

Comment:
You could add it inside IsWindows if below (https://github.com/Microsoft/msbuild/pull/759/files?diff=unified#diff-adf408463d779e2551ad431a97da518dR3291). It seems to already be including a windows only invalid string.

Or is the registry not available on xplat at all, regardless of OS? In that case you're right, Theory with ifdef would be the only way, but not worth the pain right now.